### PR TITLE
Draft: [onert] support ReLU6 training

### PR DESCRIPTION
**Pull request by zetwhite**
This draft is to support ReLU6 training feature.

 - Add ReLU6Grad cker
 - Extract fused activation gradient calculation part to reuse
 - Add Relu6 to EW activation layer 

ONE-DCO-1.0-Signed-off-by: SeungHui Youn <sseung.youn@samsung.com>

related : https://github.com/Samsung/ONE/issues/12388

### CODE REVIEW

File: compute/cker/include/cker/train/operation/ReLU6.h [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+                      const Shape &grad_shape, float *grad_data)
+{
+  const auto output_map = MapAsVector(output_data, output_shape);
+  const auto incoming_map = MapAsVector(incoming_data, incoming_shape);
+  auto grad_map = MapAsVector(grad_data, grad_shape);
+
+  if (output_shape == incoming_shape && output_shape == grad_shape)
+    grad_map.array() =
+      incoming_map.array() *
+      (0.0f < output_map.array() && output_map.array() < 6.0f).template cast<float>();
```
Comment (zetwhite) [inline compute/cker/include/cker/train/operation/ReLU6.h:42]: (note)

`array()` has no cost - it just change the view of data, for element-wise operation

https://eigen.tuxfamily.org/dox/group__QuickRefPage.html#:~:text=Recall%20that%20.array()%20has%20no%20cost%2C%20it%20only%20changes%20the%20available%20API%20and%20interpretation%20of%20the%20data

File: runtime/onert/backend/train/ops/OperationUtils.cc [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+                                   getShape(input_backprop), getBuffer<float>(input_backprop),
+                                   getShape(output_backprop), getBuffer<float>(output_backprop));
+      res = output_backprop;
+      break;
+    default:
+      throw std::runtime_error("Unsupported activation type yet");
+  }
+
+  return res;
+}
```
Comment (zetwhite) [inline runtime/onert/backend/train/ops/OperationUtils.cc:60]: @ragmani 

While I was supporting `relu6`, 
I found out that each layer(Convolution, FullyConnected and Pool for now) implements activation backpropagation on its own. 
 
I'd like to extract fused activation's backpropagation part as OperationUtil to remove code duplication. 

But before going on, I'd like to hear your opinion. Do you think it is fine?  
( I'm a bit afraid that there's a reason it has the current implementation/structure. )

Comment (ragmani) [reply runtime/onert/backend/train/ops/OperationUtils.cc:60]: Sorry for the late reply. Perhaps, I missed your comment.

Comment (zetwhite) [reply runtime/onert/backend/train/ops/OperationUtils.cc:60]: No problem! 
I didn't ask again, because it wasn't in a hurry at all.

**Comment by zetwhite** [comment]
related code is all merged. so close this Draft :)

**Pull request closed without merging**